Mathlib/Data/List/Basic.lean

```diff
@@ -994,12 +994,13 @@ theorem filterMap_eq_map_iff_forall_eq_some {f : α → Option β} {g : α → 
     l.filterMap f = l.map g ↔ ∀ x ∈ l, f x = some (g x) where
   mp := by
     induction l with | nil => simp | cons a l ih => ?_
-    rcases ha : f a with - | b <;> simp [ha]
-    · intro h
+    rcases ha : f a with - | b
+    · simp only [ha, filterMap_cons_none, map_cons, mem_cons, forall_eq_or_imp, reduceCtorEq,
+        false_and, imp_false]
+      intro h
```

Comment:
```suggestion
    · intro h
      simp [ha] at h
```
This works too, and satisfies the linter because it only stains `h`, not the goal.
Alternatively,
```
    · intro h
      have : (filterMap f l).length = l.length + 1 := by grind
      grind
```
works for the whole block.
Which do you like best?

Reply:
I like the last version best.